### PR TITLE
EOS-20528 : S3 Copy objects STs to verify the sanity of the objects.

### PR DESCRIPTION
**Pull request by pranavPawar07**
Signed-off-by: root <root@ssc-vm-4852.colo.seagate.com>

### CODE REVIEW

File: st/clitests/awss3api_spec.py [hunk 1]
```diff
@@ -110,6 +112,119 @@ def create_object_list_file(file_name, obj_list=[], quiet_mode="false"):
...
+    key = root_file
+    with open(file_to_create, 'wb+') as fout:
+        fout.write(os.urandom(filesize[start]))
+    start = start + 1
+
+#calculating md5sum in dict.
+md5sum_dict_before_upload = {}
+for file in file_list_before_upload:
+    path = os.path.join(upload_dir, file)
+    md5sum_dict_before_upload[file] = get_md5_sum(path)
```
Comment (nileshgovande) [inline st/clitests/awss3api_spec.py:144]: Combine two loops

File: st/clitests/awss3api_spec.py [hunk 2]
```diff
@@ -110,6 +112,119 @@ def create_object_list_file(file_name, obj_list=[], quiet_mode="false"):
...
+AwsTest('Aws Upload folder recursively').upload_objects("copyobjectbucket", upload_dir)\
+        .execute_test().command_is_successful()
+
+# Create a new destination bucket.
+AwsTest('Aws can create destination bucket for CopyObject API')\
+    .create_bucket("destination-bucket")\
+    .execute_test().command_is_successful()
+
+#Copy objects to destination bucket.
+AwsTest('Aws can copy object to different bucket')\
```
Comment (nileshgovande) [inline st/clitests/awss3api_spec.py:156]: Add in loop

File: st/clitests/awss3api_spec.py [hunk 3]
```diff
@@ -110,6 +112,119 @@ def create_object_list_file(file_name, obj_list=[], quiet_mode="false"):
...
+AwsTest('Aws can Download folder recursively').download_objects("destination-bucket", download_dir)\
+        .execute_test().command_is_successful()
+
+#getting files from download_dir.
+file_list_after_download = os.listdir(download_dir)
+
+#calculating md5sum after downloading the objects.
+md5sum_dict_after_download = {}
+for file in file_list_after_download:
+    path = os.path.join(download_dir, file)
```
Comment (nileshgovande) [inline st/clitests/awss3api_spec.py:182]: Combine two loops

File: st/clitests/awss3api_spec.py [hunk 4]
```diff
@@ -110,6 +112,119 @@ def create_object_list_file(file_name, obj_list=[], quiet_mode="false"):
...
+
+#deleting objects from source bucket.
+AwsTest('Aws can delete 1kfile from source bucket').delete_object("copyobjectbucket", "1kfile")\
+    .execute_test().command_is_successful()
+
+AwsTest('Aws can delete 5kfile from source bucket').delete_object("copyobjectbucket", "5kfile")\
+    .execute_test().command_is_successful()
+
+AwsTest('Aws can delete 32kfile from source bucket').delete_object("copyobjectbucket", "32kfile")\
+    .execute_test().command_is_successful()
```
Comment (nileshgovande) [inline st/clitests/awss3api_spec.py:204]: src and dst objects can be deleted in a loop

File: st/clitests/awss3api_spec.py [hunk 5]
```diff
@@ -110,6 +112,119 @@ def create_object_list_file(file_name, obj_list=[], quiet_mode="false"):
...
+    md5sum_dict_after_download[file] = get_md5_sum(path)
+
+print("***************Checksum test start****************")
+corruption = False
+for file in md5sum_dict_before_upload:
+    if md5sum_dict_before_upload[file] != md5sum_dict_after_download[file]:
+        corruption = True
+        break
+if corruption:
+    print("***************Checksum test Failed****************")
```
Comment (nileshgovande) [inline st/clitests/awss3api_spec.py:192]: Use assert to indicate failure

Comment (nileshgovande) [reply st/clitests/awss3api_spec.py:192]: Move the comparison logic after cleanup

File: st/clitests/awss3api_spec.py (hunk 3 above)
Comment (pranavPawar07) [reply st/clitests/awss3api_spec.py:175]: done

File: st/clitests/awss3api_spec.py (hunk 4 above)
Comment (pranavPawar07) [reply st/clitests/awss3api_spec.py:184]: done

File: st/clitests/awss3api_spec.py (hunk 1 above)
Comment (pranavPawar07) [reply st/clitests/awss3api_spec.py]: done

File: st/clitests/awss3api_spec.py (hunk 2 above)
Comment (pranavPawar07) [reply st/clitests/awss3api_spec.py]: done

File: st/clitests/awss3api_spec.py (hunk 5 above)
Comment (pranavPawar07) [reply st/clitests/awss3api_spec.py]: done

**Review by vimalraghwani20 — APPROVED** [review]

**Comment by pranavPawar07** [comment]
Pre merge : http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/1319/console
Mini Prov : http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Mini-Provisioning/job/cortx-s3server/1220/console

**Comment by nileshgovande** [comment]
Premerge pass - http://eos-jenkins.colo.seagate.com/job/Cortx-PR-Build/job/Component-Test/job/S3Server-PreMerge-Test/1319/

**Pull request merged**